### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by Slagrimas**
Took out hardcoded id's from backend and adjusted components to be responsive with their id.

### CODE REVIEW

File: src/app/pages/customerProfile/customerProfile.component.ts [hunk 1]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
 })
 
 export class CustomerProfileComponent implements OnInit {
+    urlId: string;
     isLoggedIn: boolean = false;
     loginPressed: boolean = false;
     editClicked: boolean = true;
-    customer: object;
+    customer: any;
+    correctCustomer: boolean = false;
```
Comment (taesup) [inline src/app/pages/customerProfile/customerProfile.component.ts]: The variable name here isn't descriptive enough. After reading through your code it seems like something more akin to ownProfile would be a more apt variable name.

File: src/app/pages/editVendor/editVendor.component.ts [hunk 2]
```diff
@@ -2,13 +2,18 @@ import { Component, OnInit } from '@angular/core';
...
 
 @Component({
     templateUrl: './editVendor.component.html',
     styleUrls: ['./editVendor.component.scss']
 })
 
 export class EditVendorComponent implements OnInit {
+    urlId: string;
+    vendor: any;
+    correctVendor: boolean = false;
```
Comment (taesup) [inline src/app/pages/editVendor/editVendor.component.ts]: Same note here with regard to the name of this variable.

File: src/app/pages/home/home.component.ts [hunk 3]
```diff
@@ -9,42 +9,42 @@ import { Router } from '@angular/router';
...
-        state:'',
-        zip_code:'',
-        email:'',
-        customer_id:null,
-        customerId:null,
-        can_bid:false,
-        budget:null,
-        category_id: null,
-      }
+    post: any
```
Comment (taesup) [inline src/app/pages/home/home.component.ts]: You really need to stop doing this.

File: src/app/pages/new-message/new-message.component.ts [hunk 4]
```diff
@@ -10,23 +10,14 @@ import { SessionService } from 'src/app/services/session.service';
...
 })
 export class NewMessageComponent {
-  user: {
-    username: string;
-  } = {
-      username: "annieL",
-    };
-
-
+  vendor: any;
```
Comment (taesup) [inline src/app/pages/new-message/new-message.component.ts]: You should avoid using any at all times. Describe the object using typescript

File: src/app/pages/postDetail/postDetail.component.ts [hunk 5]
```diff
@@ -8,39 +8,40 @@ import { Router, ActivatedRoute, ParamMap } from '@angular/router';
...
-      city: '',
-      state: '',
-      zip_code: '',
-      email: '',
-      customer_id: null,
-      customerId: null,
-      can_bid: false,
-      budget: null
-    }
+post: any;
```
Comment (taesup) [inline src/app/pages/postDetail/postDetail.component.ts]: Please stop doing this.

File: src/app/pages/profile/profile.component.ts [hunk 6]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
...
-    constructor(private backend: BackendService, private router: Router, private session: SessionService) {
-        this.backend.getVendor()
-        .then((data) => {
-            this.vendor = data
-        })
-        .catch((err) => {
-            return this.router.navigate(['/error']);
-        })
+    vendor: any;
+    correctVendor: boolean = false;
```
Comment (taesup) [inline src/app/pages/profile/profile.component.ts]: Same note with regard to the variable name

File: src/app/pusher.service.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ export class PusherService {
   }
 
   initializePusher(): void {
-    this.pusher = new Pusher(environment.pusher.key, { authEndpoint: '/pusher/auth' });
+    this.pusher = new Pusher(environment.pusher.key, { authEndpoint: 'http://localhost:8989/pusher/auth' });
```
Comment (taesup) [inline src/app/pusher.service.ts]: Why was this added back? You should avoid hardcoding in your domain as much as possible.

File: src/app/services/backend.service.ts [hunk 8]
```diff
@@ -99,13 +99,13 @@ export class BackendService {
...
-    const url = this.baseUrl + `api/customers/2`
+  getCustomer(id) {
+    const url = this.baseUrl + `api/customers/${id}`;
     return this.http.get(url).toPromise()
   }
 
-  editCustomer(data) {
-    const userUrl = this.baseUrl + `api/customers/2/edit`;
+  editCustomer(data, id) {
+    const userUrl = this.baseUrl + `api/customers/${id}/edit`;
```
Comment (taesup) [inline src/app/services/backend.service.ts]: So it looks like the data param here already has an id as a property. Don't pass that id in again as a second parameter, you could just as easily used `${data.id}`

File: src/environments/environment.ts [hunk 9]
```diff
@@ -2,12 +2,10 @@
...
 // The list of file replacements can be found in `angular.json`.
 
-require('dotenv').config()
-
-const IAM_PUSHER_KEY = process.env.pusher_key
 export const environment = {
+  production: false,
   pusher: {
-    key: IAM_PUSHER_KEY,
+    key: '<YOUR_PUSHER_KEY>',
```
Comment (taesup) [inline src/environments/environment.ts]: Why is this not using the value from process.env?

File: src/main.ts [hunk 10]
```diff
@@ -5,7 +5,7 @@ import { AppModule } from './app/app.module';
 import { environment } from './environments/environment';
 
 
-if (environment.production) {
+if (environment) {
```
Comment (taesup) [inline src/main.ts]: Again, don't edit this file.

**Review by taesup — COMMENTED** [review]
Lots of stuff to fix.

File: src/app/pages/customerProfile/customerProfile.component.ts [hunk 11]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
 })
 
 export class CustomerProfileComponent implements OnInit {
+    urlId: string;
     isLoggedIn: boolean = false;
     loginPressed: boolean = false;
     editClicked: boolean = true;
-    customer: object;
+    ownProfile: any;
+    correctCustomer: boolean = false;
```
Comment (taesup) [inline src/app/pages/customerProfile/customerProfile.component.ts]: You changed the one variable. CorrectCustomer makes no sense, ownProfile makes more sense. You should change back ownProfile to customer, and change correctCustomer to ownProfile.

Comment (taesup) [reply src/app/pages/customerProfile/customerProfile.component.ts]: Also, stop using any.

File: src/app/pages/edit/edit.component.ts [hunk 12]
```diff
@@ -9,6 +9,8 @@ import { Router } from '@angular/router';
 })
 
 export class EditComponent implements OnInit {
+    id: any;
```
Comment (taesup) [inline src/app/pages/edit/edit.component.ts]: No any allowed. Write out datatype it is in typescript.

File: src/app/pages/edit/edit.component.ts [hunk 13]
```diff
@@ -9,6 +9,8 @@ import { Router } from '@angular/router';
...
+    customer: any;
```
Comment (taesup) [inline src/app/pages/edit/edit.component.ts]: No any allowed.

File: src/app/pages/editVendor/editVendor.component.ts [hunk 14]
```diff
@@ -2,13 +2,18 @@ import { Component, OnInit } from '@angular/core';
...
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
     templateUrl: './editVendor.component.html',
     styleUrls: ['./editVendor.component.scss']
 })
 
 export class EditVendorComponent implements OnInit {
+    urlId: string;
+    vendorProfile: any;
```
Comment (taesup) [inline src/app/pages/editVendor/editVendor.component.ts]: No any allowed.

File: src/app/pages/editVendor/editVendor.component.ts [hunk 15]
```diff
@@ -2,13 +2,18 @@ import { Component, OnInit } from '@angular/core';
...
+    correctVendor: boolean = false;
```
Comment (taesup) [inline src/app/pages/editVendor/editVendor.component.ts]: This is be renamed to ownProfile, correctVendor makes no sense.

File: src/app/pages/editVendor/editVendor.component.ts [hunk 16]
```diff
@@ -2,13 +2,18 @@ import { Component, OnInit } from '@angular/core';
...
+    id: any;
```
Comment (taesup) [inline src/app/pages/editVendor/editVendor.component.ts]: No any allowed.

File: src/app/pages/new-message/new-message.component.ts [hunk 17]
```diff
@@ -10,23 +10,26 @@ import { SessionService } from 'src/app/services/session.service';
...
+  };
+  customer: {
+    first_name: string,
+    last_name: string,
+  } = {
+    first_name: '',
+    last_name: ''
+  };
   message: string;
   data: any;
```
Comment (taesup) [inline src/app/pages/new-message/new-message.component.ts]: No any allowed.

File: src/app/pages/profile/profile.component.ts [hunk 18]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
 })
 
 export class ProfileComponent implements OnInit {
+    urlId: string;
     isLoggedIn: boolean = false;
     loginPressed: boolean = false;
     data: any;
```
Comment (taesup) [inline src/app/pages/profile/profile.component.ts]: No any allowed.

File: src/app/pages/profile/profile.component.ts [hunk 19]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
...
-
-    constructor(private backend: BackendService, private router: Router, private session: SessionService) {
-        this.backend.getVendor()
-        .then((data) => {
-            this.vendor = data
-        })
-        .catch((err) => {
-            return this.router.navigate(['/error']);
-        })
+    vendorProfile: any;
```
Comment (taesup) [inline src/app/pages/profile/profile.component.ts]: No any allowed.

File: src/app/pages/profile/profile.component.ts [hunk 20]
```diff
@@ -9,23 +10,27 @@ import { Router } from '@angular/router';
...
+    correctVendor: boolean = false;
```
Comment (taesup) [inline src/app/pages/profile/profile.component.ts]: This should be renamed to ownProfile

File: src/environments/environment.ts [hunk 21]
```diff
@@ -1,13 +1,14 @@
...
 
-require('dotenv').config()
+// const IAM_KEY = process.env.key
 
-const IAM_PUSHER_KEY = process.env.pusher_key
 export const environment = {
+  production: false,
   pusher: {
-    key: IAM_PUSHER_KEY,
+    key: 'IAM_KEY',
```
Comment (taesup) [inline src/environments/environment.ts]: Why is this a string?

**Review by taesup — COMMENTED** [review]
More feedback.

File: src/app/pages/home/home.component.ts [hunk 22]
```diff
@@ -9,42 +9,42 @@ import { Router } from '@angular/router';
...
-        state:'',
-        zip_code:'',
-        email:'',
-        customer_id:null,
-        customerId:null,
-        can_bid:false,
-        budget:null,
-        category_id: null,
-      }
+    post: any;
```
Comment (taesup) [inline src/app/pages/home/home.component.ts]: I think you missed this one? Or are you letting Jenn take care of this one?

File: src/app/pages/postDetail/postDetail.component.ts [hunk 23]
```diff
@@ -8,39 +8,40 @@ import { Router, ActivatedRoute, ParamMap } from '@angular/router';
...
-      city: '',
-      state: '',
-      zip_code: '',
-      email: '',
-      customer_id: null,
-      customerId: null,
-      can_bid: false,
-      budget: null
-    }
+  // post: {
```
Comment (taesup) [inline src/app/pages/postDetail/postDetail.component.ts:11]: This one as well.

File: src/app/pages/profile/profile.component.ts [hunk 24]
```diff
@@ -9,23 +10,56 @@ import { Router } from '@angular/router';
...
+          website: "",
+          description: "",
+          company_name: "",
+          isLoggedIn: false,
+          city: "",
+          state: "",
+          zip_code: null,
+        }
+        
+    correctVendor: boolean = false;
```
Comment (taesup) [inline src/app/pages/profile/profile.component.ts]: You should rename this to ownProfile as well

File: src/environments/environment.ts (hunk 21 above)
Comment (taesup) [inline src/environments/environment.ts]: Again, why is this a string? You should be using the `IAM_KEY` value here. Along with the dotenv and process.env.

File: src/app/pages/home/home.component.ts (hunk 22 above)
Comment (Slagrimas) [reply src/app/pages/home/home.component.ts]: Jenn will take care of these

File: src/app/pages/postDetail/postDetail.component.ts (hunk 23 above)
Comment (Slagrimas) [reply src/app/pages/postDetail/postDetail.component.ts:11]: Jenn will take care of these

**Review by taesup — APPROVED** [review]
